### PR TITLE
Code style improvements

### LINKED ISSUE #175: The body of an if should be on a separate line
#### Broken rules:
See title. Having the body of the if on the same line make code more difficult to read for no added benefit. See example below:

image

@FraserGreenroyd , FYI. As it was more an implicit rule based on common sense than anything, it might be good to have it explicitly stated and checked.

### LINKED ISSUE #193: Improve coding style to better align with the rest of the BHoM
See for example BHoM_UI for reference of proper code style.

A few things I have noticed: 
- Public/private fields/properties/methods/constructors are all mixed together without clear separation (see for example [AddIn.cs](https://github.com/BHoM/Excel_Toolkit/blob/master/Excel_UI/Addin/AddIn.cs))
- Variable, method arguments and private properties do no follow the proper casing (all variables/argument are `camelCased`, methods are `PascalCased`)
- Inconsistent naming conventions for private variables (should always be `m_SomeName`)
- Methods needs to be separated by comment line and sections separated by 3 line comments
- Avoid clustering of code on a single line (see #173)

### PR DESCRIPTION
**Pull request by awakeman**
### Issues addressed by this PR

Closes #175 
Closes #193 

- Moved all one-line if statement bodies to new line. (#175)
- Reordered Methods/Constructors/Properties/Fields into groups (#193)
- Added headers for groups (#193)
- Added method separators (#193)

### Test files

N/A

### CODE REVIEW

File: Excel_UI/Addin/AddIn.cs [hunk 1]
```diff
@@ -356,70 +431,36 @@ private static bool IsNullMissingOrEmpty(object obj)
...
+                    InitCallers();
                 return m_formulea;
             }
         }
+
+        /*****************************************************************/
+        /******* Private fields                             **************/
+        /*****************************************************************/
+
+        private Dictionary<string, CallerFormula> m_formulea;
```
Comment (adecler) [inline Excel_UI/Addin/AddIn.cs]: Incorrect variable naming. Please re-read details in issue #193

Comment (adecler) [reply Excel_UI/Addin/AddIn.cs]: Same for all other private variables

File: Excel_UI/UI/Callers/Explode.cs [hunk 2]
```diff
@@ -50,5 +46,15 @@ public ExplodeCaller() : base(typeof(Methods.Properties).GetMethod("Explode"))
         public override int GroupIndex => m_native.GroupIndex;
 
         private UI.Components.ExplodeCaller m_native = new UI.Components.ExplodeCaller();
+
```
Comment (adecler) [inline Excel_UI/UI/Callers/Explode.cs:77]: Incorrect format for the properties

File: Excel_UI/UI/Components/Adapter/CreateAdapter.cs [hunk 3]
```diff
@@ -49,6 +49,8 @@ public class CreateAdapterFormula : CallerFormula
         /*******************************************/
 
         public CreateAdapterFormula() : base() { }
+
+        /*******************************************/
```
Comment (adecler) [inline Excel_UI/UI/Components/Adapter/CreateAdapter.cs:51]: - Incorrect format for first property

Comment (adecler) [reply Excel_UI/UI/Components/Adapter/CreateAdapter.cs:51]: Same in other files

**Review by adecler — CHANGES_REQUESTED** [review]
- Going in the right direction for Excel_UI although there are still a few problems and you have skipped a couple of files. Also inconsistent format for your separator comments on some of your files.

- Nothing done on Excel_Engine

**Comment by awakeman** [comment]
Spoke with @al-fisher and we agreed it would be expedient to allow for someone other than @adecler to pick up this review in his absence as this is otherwise blocking other tasks on this toolkit. obviously reviewer should review against @adecler's comments to check they are appropriately addressed.

File: Excel_UI/UI/Templates/CallerFormula.cs [hunk 4]
```diff
@@ -35,25 +35,23 @@ namespace BH.UI.Excel.Templates
 {
     public abstract class CallerFormula
     {
-        private IExcelSelectorMenu m_menu;
+        private IExcelSelectorMenu m_Menu;
```
Comment (FraserGreenroyd) [inline Excel_UI/UI/Templates/CallerFormula.cs]: Should this be at the bottom of the file?

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]
I did a review against:
 - The original issues this PR aims to solve
 - The comments made by @adecler 

I will admit, this is a little tricky for me as I've been out of the loop on any discussions. However, in the interest of allowing this PR to move forward and unblock you @awakeman  I have made only one suggestion - if the suggestion is invalid, please let me know and I am happy to then give an approving review instead.

I acknowledge that there may be other coding improvements which need to be made, and I may have missed some of the finer points which @adecler would be after in this PR. However, I believe the pragmatic view, given the current circumstances, is to allow this PR to merge (pending decision on my requested change) and allow @adecler to review again on his return and raise new issues as appropriate. I hope this will be acceptable to all concerned :smile:

**Review by adecler — DISMISSED** [review]
Thanks @FraserGreenroyd for picking this up and apologies for my absence.

Still a few things left to do on top of the comment from Fraser:

- In `AddIn.FlagUsed()`, `Wb` should be `wb`. Even better would be to use a more explicit name. Same in `Project.SaveData()`
- `Properties.Transpose()` has no separator and should in the private section.
- In `Properties.GetPropertyDictionaries()`, `else` should be on a new line. Same for `Project.GetId()` and `Project.Deserialise()`
- `Explode.m_Native` should be in the private field section
- `SingleOptionCallerFormula.MenuRoot` incorrect section and format.
- ArrayResizer` and `DebugConfig` not done

A few things on compliance as well. Although I am happy for those to be done on a separate PR as it was not part of this scope but thought I would mention it while I noticed them.

- Inconsistencies with file naming vs method naming (e.g. `FromExcel` vs `ToBHoM`, `Note` vs `SetNote`, `TextRef` vs `RefText`)
- Incorrect naming of `Query` methods: should not start with `Get`
- `Timer` should be in an `Objects` folder

**Comment by awakeman** [comment]
> * In `AddIn.FlagUsed()`, `Wb` should be `wb`. Even better would be to use a more explicit name. Same in `Project.SaveData()`

Fixed
> * `Properties.Transpose()` has no separator and should in the private section.

Fixed
> * In `Properties.GetPropertyDictionaries()`, `else` should be on a new line. Same for `Project.GetId()` and `Project.Deserialise()`

Fixed
> * `Explode.m_Native` should be in the private field section

Fixed
> * `SingleOptionCallerFormula.MenuRoot` incorrect section and format.

Fixed
> * ArrayResizer`and`DebugConfig` not done

Fixed


> A few things on compliance as well. Although I am happy for those to be done on a separate PR as it was not part of this scope but thought I would mention it while I noticed them.
> 
> * Inconsistencies with file naming vs method naming (e.g. `FromExcel` vs `ToBHoM`, `Note` vs `SetNote`, `TextRef` vs `RefText`)
> * Incorrect naming of `Query` methods: should not start with `Get`
> * `Timer` should be in an `Objects` folder

This will be addressed in a compliance PR once this is merged

**Review by FraserGreenroyd — APPROVED** [review]
Have gone through this again in the spirit of unblocking for more PRs and improvements. I am happy that @adecler second changes appear to be resolved so am approving this for merging to unblock future work. I accept I may have missed something @adecler may find, but would recommend we go for a new issue to resolve that to allow this to move forward with Excel UI.

**Pull request merged**